ebmdatalab/maps.py

```diff
@@ -1,131 +1,124 @@
-
-    # Join map with data
-    gdf = ccgs.join(df, rsuffix="_orig")
-
-    # Split into london and rest of England
-    gdf_london = gdf[gdf["is_london"] == True]
-    gdf_roe = gdf[gdf["is_london"] == False]
-
-    # set common value limits for colour scale
-    default_plot_options = {
-        'vmin': df[column].min(),
-        'vmax': df[column].max(),
-        'edgecolor': "black",
-        'linewidth': 0.1,
-        'cmap': "OrRd",
-    }
-
-    if plot_options is None:
-        plot_options = {}
-
-    for k, v in default_plot_options.items():
-        if k not in plot_options:
-            plot_options[k] = v
-
-    def plot(gdf, ax, title="", legend=True):
-        gdf.plot(
-            ax=ax,
-            column=column,
-            legend=legend,
-            **plot_options
-        )
-        ax.set_aspect(1.63)
-        if title:
-            ax.set_title(title, size=14)
-        ax.axis("off")
-
-    fig = plt.gcf()
-    if not subplot_spec:
+
+    def __init__(self, description=None, author="EBM DataLab", *args, **kwargs):
+        """Map Loader is initiated with description and author for clarity
+
+            Args:
+                description (str): Description at a broad level of the dataset
+                author (str): Author - set as EBM Datalab if left blank
+
+        """
+
+        self.description = description
+        self.author = author
+
+        self.date = datetime.datetime.now()
+        """Datetime (obj): Automatically records datetime as time class is initiated"""
+
+        self.plots = {}
+        """dict: initally blank to allow plots to be saved"""
+
+    def load_data(self, data, mapping_data_path, index_file_path=None):
+        """
+        Loads data from a pandas dataframe and combines with a mapping json file
+
+        :param data: pd.dataframe file
+        :param mapping_data_path (str): the path of the json file
+        :param index_file_path (str): path of the index file. Index file contains additionally information
+                about the geographical areas such as London False/True
+
+        :return: self.final_df
+        """
+
+        self.map_data = gpd.read_file(mapping_data_path)
+        if index_file_path == None:
+            final = gpd.GeoDataFrame(
+                pd.merge(left=data, right=self.map_data, how='inner', left_on='row_id', right_on='code'))
+            self.final_df = final
+        else:
+            self.index_file = pd.read_csv(index_file_path)
+            interim = gpd.GeoDataFrame(
+                pd.merge(left=data, right=self.index_file, how='inner', left_on='row_id', right_on='code',
+                         left_index=True))
+            final = gpd.GeoDataFrame(
+                pd.merge(left=interim, right=self.map_data, how='inner', left_on='row_id', right_on='code'))
+            self.final_df = final
+
+    def create_map(self, plot_value, title, description=None,
+                   london_separate=True,
```

Comment:
It looks like almost the entire body of this function switches on the value of `london_separate`. This suggests it might be simpler to have two functions `create_map` and `create_map_with_london_separate`.

It's a topic that gets written about quite a lot ([here's](https://www.martinfowler.com/bliki/FlagArgument.html) a reasonably short article on it) but the basic idea is that removing conditionals like this makes code easier to reason about and harder to use in the wrong way.

(I'm aware that this was inherited from the previous implementation, by the way, but I thought the general point was worth raising.)

Reply:
this is a good idea. i will try this.